tool/src/js/layout.rs

```diff
@@ -57,24 +117,32 @@ pub fn unit_size_alignment() -> Layout {
 
 /// Get the [`Layout`] for a specific type.
 pub fn type_size_alignment<P: hir::TyPosition>(typ: &Type<P>, tcx: &TypeContext) -> Layout {
+    type_size_alignment_and_scalar_count(typ, tcx).0
+}
+
+/// Get the [`Layout`] for a specific type, as well as the number of scalar fields it contains
+pub fn type_size_alignment_and_scalar_count<P: hir::TyPosition>(
+    typ: &Type<P>,
+    tcx: &TypeContext,
+) -> (Layout, usize) {
     match typ {
         // repr(C) fieldless enums use the default platform representation: isize
-        Type::Enum(..) => Layout::new::<usize_target>(),
-        Type::Opaque(..) => opaque_size_alignment(),
-        Type::Slice(..) => Layout::new::<(usize_target, usize_target)>(),
-        Type::Primitive(p) => primitive_size_alignment(*p),
+        Type::Enum(..) => (Layout::new::<usize_target>(), 1),
+        Type::Opaque(..) => (opaque_size_alignment(), 1),
+        Type::Slice(..) => (Layout::new::<(usize_target, usize_target)>(), 2),
```

Comment:
Thought: Okay, "scalar" doesn't mean "a primitive value", it means "number of u32s". ?

Reply:
No, it's integral primitives. Slices aren't primitives.